src/org/exist/xmldb/LocalCollection.java

```diff
@@ -249,7 +249,7 @@ public int getChildCollectionCount() throws XMLDBException {
 
     @Override
     public String getName() throws XMLDBException {
-        return this.<String>read().apply((collection, broker, transaction) -> collection.getURI().toString());
```

Comment:
This will throw an `XMLDBException(ErrorCodes.INVALID_COLLECTION, ...` if the `Collection` was removed from the server, whereas after your change it is possible to call `getName()` on a Collection which no longer exists.